corehq/apps/app_manager/views/utils.py

```diff
@@ -672,3 +674,14 @@ def handle_custom_assertions(custom_assertions_string, form, lang):
         new_assertions.append(new_assertion)
 
     form.custom_assertions = new_assertions
+
+
+def capture_user_errors(view_fn):
+    """Catches AppMisconfigurationError and returns the message as a 400"""
+    @wraps(view_fn)
+    def inner(request, *args, **kwargs):
+        try:
+            return view_fn(request, *args, **kwargs)
+        except AppMisconfigurationError as e:
+            return json_response({'message': str(e)}, status_code=400)
```

Comment:
The three view functions I'm working in each use a mix of `json_response({'message': ...})` and `HttpResponseBadRequest(message)`.  Testing shows that both seem to work about the same for the form and module views (in that the user-facing alert shows only the message), but only the former works for `edit_app_attr`, since it's response is captured by `edit_commcare_settings`.

Reply:
thanks for the background - can do!